csharp/src/Ice/Internal/StreamSocket.cs

```diff
@@ -443,6 +461,7 @@ private void ioCompleted(object sender, SocketAsyncEventArgs e)
         switch (e.LastOperation)
         {
             case SocketAsyncOperation.Receive:
+                _pendingRead.Set();
```

Comment:
Why is this set here and not in startRead when it doesn't complete synchronously?

Reply:
_pendingRead is about "pending" reads / "is waiting to be read". If the read completes synchronously, the bytes have been read and the IdleTimeoutTransceiverDecotator has already rescheduled the idle check a few instructions before:

https://github.com/zeroc-ice/ice/blob/765d81587ad71831d35f64f47b5460dcfc8b3d08/csharp/src/Ice/Internal/IdleTimeoutTransceiverDecorator.cs#L73